src/handler/sign_tx.c

```diff
@@ -98,10 +101,32 @@ int handler_sign_tx(buffer_t *cdata, uint8_t chunk, bool more) {
                 return ui_display_transaction();
```

Comment:
This function (not just this snippet but I can't place the comment higher) is too long and too complicated, it should be split